tests/apps/binding_input/app.go

```diff
@@ -0,0 +1,103 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"sync"
+
+	"github.com/gorilla/mux"
+)
+
+const appPort = 3000
+
+type messageBuffer struct {
+	lock     *sync.RWMutex
+	messages []string
+}
+
+func (m *messageBuffer) add(message string) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	m.messages = append(m.messages, message)
+}
+
+func (m *messageBuffer) getAll() []string {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	return m.messages
+}
+
+var messages messageBuffer = messageBuffer{
+	lock:     &sync.RWMutex{},
+	messages: []string{},
+}
+
+type indexHandlerResponse struct {
+	Message string `json:"message,omitempty"`
+}
+
+type testHandlerResponse struct {
+	ReceivedMessages []string `json:"received_messages,omitempty"`
+}
+
+// indexHandler is the handler for root path
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	log.Println("indexHandler is called")
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(indexHandlerResponse{Message: "OK"})
+}
+
+func testTopicHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodOptions {
+		log.Println("test-topic binding input has been accepted")
+		w.WriteHeader(http.StatusOK)
+		return
+	}
+
+	var message string
+	err := json.NewDecoder(r.Body).Decode(&message)
+	if err != nil {
+		log.Printf("error parsing test-topic input binding payload: %s", err)
+		w.WriteHeader(http.StatusOK)
```

Comment:
Some other status, please - this is an error condition.

Reply:
@artursouza ahh, I forgot that I wanted to ask this. If I return some error status here, wouldn't it attempt to redeliver the message?

Reply:
Good point. You are right, there is no concept of NonRetriable error for bindings in Dapr - same problem with PubSub: https://github.com/dapr/docs/tree/master/howto/trigger-app-with-input-binding

Please, add comment in the code explaining why it is http.StatusOK so someone else (like me) does not make the mistake of "fixing" it.

I have a proposal to enhance the PubSub API and features in Dapr and adding the classification of retriable vs non-retriable errors is in the list since it can allow the deadletter feature. On the other hand, I am not sure if this retriable/non-retriable error code is beneficial to Bindings as well. You are free to start a proposal for that and see what the community thinks.

Reply:
Done